File: Paint_FrontEnd/src/app/Shapes/circle.ts
import { IShape } from "./ishape";

export class Circle extends IShape{
    private _radius: number;
    
    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean, radius: number){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._radius = radius;
        this.type = "circle";
    }

    public get radius(): number {
        return this._radius;
    }
    public set radius(value: number) {
        this._radius = value;
    }
}

File: Paint_BackEnd/src/main/java/com/paint/Paint_BackEnd/Shapes/Circle.java
package com.paint.paint_backend.Shapes;

public class Circle extends IShape{
    private double _radius;
    Circle(int x, int y, String stroke, double strokeWidth, String fill, int rotate, boolean draggable, String id, boolean update, boolean deleteflag, double radius){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._radius = radius;
        super.set_type("circle");
    }

    public double get_radius() {
        return _radius;
    }

    public void set_radius(double _radius) {
        this._radius = _radius;
    }
}
